plugin.rb

```diff
@@ -185,7 +185,7 @@ def find_topic(topic_id, post_number)
   end
 
   DiscourseEvent.on(:post_created) do |post|
-    if SiteSetting.slack_enabled?
+    if SiteSetting.slack_enabled? && DiscourseSlack::Slack.guardian.can_see?(post)
```

Comment:
I'll move the logic into 

https://github.com/discourse/discourse-slack-official/blob/97ceba2ebda7f608ea9d6e9c8d83451ffc431589/lib/discourse_slack/slack.rb#L156-L157